### PR TITLE
Fix Full screen mirror behavior

### PR DESCRIPTION
**Pull request by samcake**
- Solve the mirror rendering for the main scene render like for the rear view mirror by blitting and flipping the pixels horizontally

- Revisit the way the camera  position and orientation are evaluated in Fullscreen  AND HMD mode. 
- No more knowledge of the mirror mode needed in MyAvatar

- Re-fix the screwed ProceduralSkybox code by actually committing my changes to the vertex shader
Don't forget to test in "entry" with a procedural skybox

- The avatar name are displayed mirrored, we could do something about it if we think it s needed.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: ed2d45d0b6d524e5afd9755ec7a7b6416daf910f available at: https://gist.github.com/fdc66bbb74962889e3a9<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/High+Fidelity-PR-6652-win64.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6652.dmg

**Comment by ZappoMan** [comment]
This is awesome! This appears to be a TRUE mirror! So much better than the old hacky version... One thing I noticed (which likely would be hard to fix)... the Display Names over my head are mirrored...

**Comment by ZappoMan** [comment]
Another bonus of this PR fix is that desktop mirror is ALSO correct... If using DDE or Faceshift and you hold your left ARM up in the air (with hydra) and lean your head toward your left ARM, you actually see yourself lean toward your upstretched arm... AND... people see you leaning to the left...

File: interface/src/Application.cpp [hunk 1]
```diff
@@ -1326,9 +1342,77 @@ void Application::paintGL() {
...
+            if (displayPlugin->isStereo()) {
+                gpu::Vec4i srcRect;
+                srcRect.z = size.width();
+                srcRect.w = size.height();
+                gpu::Vec4i destRect;
+                destRect.x = size.width();
+                destRect.y = 0;
+                destRect.z = 0;
+                destRect.w = size.height();
+               // batch.setFramebuffer(finalFramebuffer);
```
Comment (ZappoMan) [inline interface/src/Application.cpp]: dead code - please deleted

File: interface/src/Application.cpp [hunk 2]
```diff
@@ -1231,15 +1231,28 @@ void Application::paintGL() {
             }
         } else if (_myCamera.getMode() == CAMERA_MODE_MIRROR) {
             if (isHMDMode()) {
+                auto mirrorBodyOrientation = myAvatar->getWorldAlignedOrientation() * glm::quat(glm::vec3(0.0f, PI + _rotateMirror, 0.0f));
```
Comment (ZappoMan) [inline interface/src/Application.cpp]: I have to admit, I don't understand what this has to do with HMD mode... it seems like the head pose is the head pose... I mean, this works in HMD *AND* in desktop with the face tracker, but I don't understand why it works in both those modes... why do we need to do this extra HMD stuff, when we don't appear to need to do it for desktop head tracker?

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9f1f0798cb2f38b17dfa0aa50ab1abe152e9eca5 available at: https://gist.github.com/fdc66bbb74962889e3a9<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/High+Fidelity-PR-6652-win64.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6652.dmg

File: interface/src/avatar/MyAvatar.cpp [hunk 3]
```diff
@@ -441,14 +441,14 @@ void MyAvatar::updateFromTrackers(float deltaTime) {
         estimatedPosition = tracker->getHeadTranslation();
         _trackedHeadPosition = estimatedPosition;
         estimatedRotation = glm::degrees(safeEulerAngles(tracker->getHeadRotation()));
-        if (qApp->getCamera()->getMode() == CAMERA_MODE_MIRROR) {
+      /*  if (qApp->getCamera()->getMode() == CAMERA_MODE_MIRROR) {
```
Comment (ZappoMan) [inline interface/src/avatar/MyAvatar.cpp]: more dead code

File: interface/src/avatar/MyAvatar.cpp [hunk 4]
```diff
@@ -493,9 +493,9 @@ void MyAvatar::updateFromTrackers(float deltaTime) {
     // NOTE: this is kinda a hack, it's the same hack we use to make the head tilt. But it's not really a mirror
     // it just makes you feel like you're looking in a mirror because the body movements of the avatar appear to
     // match your body movements.
-    if ((inHmd || inFacetracker) && qApp->getCamera()->getMode() == CAMERA_MODE_MIRROR) {
+  /*  if ((inHmd || inFacetracker) && qApp->getCamera()->getMode() == CAMERA_MODE_MIRROR) {
```
Comment (ZappoMan) [inline interface/src/avatar/MyAvatar.cpp]: death to dead code

File: interface/src/avatar/MyAvatar.cpp [hunk 5]
```diff
@@ -1402,10 +1402,10 @@ void MyAvatar::updateOrientation(float deltaTime) {
         glm::vec3 euler = glm::eulerAngles(localOrientation) * DEGREES_PER_RADIAN;
 
         //Invert yaw and roll when in mirror mode
-        if (qApp->getCamera()->getMode() == CAMERA_MODE_MIRROR) {
+      /*  if (qApp->getCamera()->getMode() == CAMERA_MODE_MIRROR) {
```
Comment (ZappoMan) [inline interface/src/avatar/MyAvatar.cpp]: the zombies are among us

File: interface/src/Application.cpp (hunk 2 above)
Comment (samcake) [inline interface/src/Application.cpp]: That s a good point, let me think about that

**Comment by hifi-gustavo** [comment]
Build console output for commit: 571ccb3c206e040eb27c82c7b2293f5ec0575b6f available at: https://gist.github.com/fdc66bbb74962889e3a9<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/High+Fidelity-PR-6652-win64.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6652.dmg

File: interface/src/Application.cpp (hunk 2 above)
Comment (ZappoMan) [inline interface/src/Application.cpp]: ok - I understand - thanks for explaining.

**Comment by ZappoMan** [comment]
test this please

**Comment by samcake** [comment]
Test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 571ccb3c206e040eb27c82c7b2293f5ec0575b6f available at: https://gist.github.com/fdc66bbb74962889e3a9<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/High+Fidelity-PR-6652-win64.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6652.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 571ccb3c206e040eb27c82c7b2293f5ec0575b6f available at: https://gist.github.com/fdc66bbb74962889e3a9<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/High+Fidelity-PR-6652-win64.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6652.dmg

**Pull request merged**